nautobot/core/testing/views.py

```diff
@@ -236,6 +237,51 @@ def test_get_object_with_constrained_permission(self):
 
             return response  # for consumption by child test cases if desired
 
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+        def test_has_timestamps_and_buttons(self):
+            instance = self._get_queryset().first()
+
+            # Add model-level permission
+            obj_perm = users_models.ObjectPermission(
+                name="Test permission", actions=["view", "add", "change", "delete"]
+            )
+            obj_perm.save()
+            obj_perm.users.add(self.user)
+            obj_perm.object_types.add(ContentType.objects.get_for_model(self.model))
+
+            response = self.client.get(instance.get_absolute_url())
+
+            timestamps = f"""
+                <p class="flex-grow-1 m-0">
+                    <small class="align-items-center d-inline-flex gap-4 text-secondary">
+                        <i class="mdi mdi-bookmark-plus"></i> {date(instance.created, global_settings.DATETIME_FORMAT)}
+                        <span class="vr mx-4"></span>
+                        <i class="mdi mdi-clock"></i> {timesince(instance.last_updated)} ago
+                    </small>
+                </p>
+            """
+            self.assertBodyContains(response, timestamps, html=True)
+
+            action_buttons = [
+                f"""
+                    <a id="edit-button" class="btn btn-warning border-end-0" href="{buttons.edit_button(instance)['url']}">
+                        <span class="mdi mdi-pencil" aria-hidden="true"></span> Edit
+                    </a>
+                """,
+                f"""
+                    <a id="clone-button" class="dropdown-item" href="{buttons.clone_button(instance)['url']}">
```

Comment:
Not all models are clonable. I *think* this can be wrapped in a check like `if hasattr(instance, "clone_fields")` to fix the CI failure.

Also note that not all models are editable or deletable. That's a bit trickier, but can probably steal the logic in `nautobot/core/templatetags/buttons.py`:

```py
    object_edit_url = action_url(instance, "edit")
    object_delete_url = action_url(instance, "delete")
    object_clone_url = action_url(instance, "add")
    render_edit_button = bool(object_edit_url and can_change(context["user"], instance))
    render_delete_button = bool(object_delete_url and can_delete(context["user"], instance))
    render_clone_button = bool(
        hasattr(instance, "clone_fields") and object_clone_url and can_add(context["user"], instance)
    )
```